### DISCUSSION TITLE
How to expose the functionality of external libraries

### DISCUSSION THREAD

**Discussion by p1gp1g** [COVESA/covesa-aosp-sdk#5] [General]
There are basically 2 ways to do that:
- Exposing the dependency directly as an api dependency, and using the dependency's classes for internal process (like passing extra data for intents)
- Write new classes/wrapper that call the library

The former makes it easier to take advantage of new features, while the latter improves control over updates.

Note: the question has been raised for #3. I've used the first option, at least in a first time

**Comment by maximilian-galanis** [comment]
This is really a question of maintenance in my opinion. If this SDK is actively maintained continuously, I would prefer option 2 (to have separate wrapper classes, that use the up-connector-lib internally).
This option also has the advantage that we could set the OEM provided distributor as the default. In a car, there probably shouldn't be a choice which distributor to use.

If only very little maintenance is planned to be done, option 1 is preferable in my opinion.

Any other thoughts on this?

**Reply by paulboyes** [reply]
We should plan on this SDK being actively maintained.

**Comment by khvMX** [comment]
I believe every component of this SDK should be actively maintained. Moreover, we should probably [keep track of official maintainers](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) here.
No one was against wrapping API on the meeting today, let's stick with that.
